packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2945,6 +3016,23 @@ class InputDecoration {
   ///  * [prefixIcon], which is the same but on the leading edge.
   final Widget? suffixIcon;
 
+  /// The alignment for [suffixIcon].
+  ///
+  /// Used to align the [suffixIcon] relative to the position of the text box.
+  ///
+  /// ```dart
+  /// TextField(
+  ///   decoration: InputDecoration(
+  ///     hintText:'Top Icon Alignment',
+  ///     suffixIcon: Icon(Icons.search),
+  ///     suffixIconAlignment: TextAlignVertical.top,
+  ///   ),
+  /// )
+  /// ```
+  ///
+  /// The default value is [TextAlignVertical.center].
```

Comment:
Just to be consistent with other Flutter docs, I think this should say "Defaults to [TextAlignVertical.center]."